.circleci/config.yml

```diff
@@ -114,11 +114,18 @@ jobs:
       - checkout
       - attach_workspace:
           at: /rest
+      - run:
+          command: |
+            suffix=$(if [ <<parameters.basedir>> = "/rest" ]; then echo "REST"; else echo "AIO"; fi)
+            lower=PYPI_TOKEN_GCLOUD_$(echo $suffix)_<<parameters.cwd>>
+            upper=$(echo $lower | awk '{ print toupper }')
+            export POETRY_PYPI_TOKEN_PYPI=$(printenv | grep $upper | awk -F= '{ print $2 }')
```

Comment:
You could also write this as a grep with case insensitivity for simplicity:

```shell
suffix=$(if [ <<parameters.basedir>> = "/rest" ]; then echo "REST"; else echo "AIO"; fi)
token=$(printenv | grep -i PYPI_TOKEN_GCLOUD_${suffix}_<<parameters.cwd>> | awk -F= '{ print $2 }')
echo 'export POETRY_PYPI_TOKEN_PYPI=${token}"' >> "$BASH_ENV"
```

Reply:
Ohhh I didn't think of using case insensitivity, good idea!